### DISCUSSION TITLE
Filter a task on a specific package with `dependsOn`

### DISCUSSION THREAD

**Discussion by theolavaux** [vercel/turborepo#3183] [Help] [answered]
Setup

```
apps/
  --website (has lint and build scripts, no test for the moment / has a dependency on `packages/ui`)

packages/
   --ui (also has lint, test and build scripts)
```

I've created a `package` script (to group my `lint`, `test`, `build` scripts) in my root `package.json` that calls a related turbo task:

_package.json_
```json
{
  "workspaces": [
    "apps/*",
    "packages/*"
  ],
  "scripts": {
    "package": "turbo run package",
    "lint": "turbo run lint",
    "build": "turbo run build",   
    "test": "turbo run test",
  },
}
``` 

_turbo.json_

```json
{
  "$schema": "https://turbo.build/schema.json",
  "pipeline": {
    "lint": {
      "outputs": []
    },
    "test": {
      "outputs": []
    },
    "build": {
      "dependsOn": ["^build"],
      "outputs": [
        "dist/**",
        ".output/**",
        ".histoire/**",
        "**/.vitepress/dist/**"
      ]
    },
    "package": {
      "dependsOn": ["lint", "test", "build"]
    },
  }
}
```

When I run 
```sh
pnpm package --filter website
```

I expect the `lint`, `test` (if existing) and `build` script run for the `website` package.

Even though it's saying
```txt
• Packages in scope: website
• Running package in 1 packages
```

`lint` and `build` tasks run for both `website` AND `ui`.

It's also weird because if I acknowledge that both projects are concerned, `test` for `ui` should run but it doesn't.

**Comment by mehulkar** [comment, marked as answer]
This is mostly expected. 

- Packages in scope only lists `website`, because of the `--filter website` in your CLI command
- `build` runs for `website` because you asked it to
- `build` runs for `ui`, because `build`: `dependsOn: [^build]`, which means website's dependencies (i.e. `ui`) will also run `build`
- `test` does _not_ run for `ui` because there's nothing in the graph that causes it to run
- `lint` runs for `website` because it's part of `package`

`lint` running for `ui` seems wrong. Any chance you could make a runnable example and push it up to Github? I can debug from there
